jlm/mlir/frontend/MlirToJlmConverter.cpp

```diff
@@ -567,6 +567,33 @@ MlirToJlmConverter::ConvertOperation(
 
     return rvsdgThetaNode;
   }
+  else if (auto mlirDeltaNode = ::mlir::dyn_cast<::mlir::rvsdg::DeltaNode>(&mlirOperation))
+  {
+    auto & deltaRegion = mlirDeltaNode.getRegion();
+    auto & deltaBlock = deltaRegion.front();
+    auto terminator = deltaBlock.getTerminator();
+
+    auto mlirOutputType = terminator->getOperand(0).getType();
+    std::shared_ptr<rvsdg::Type> outputType = ConvertType(mlirOutputType);
+    auto outputValueType = std::dynamic_pointer_cast<const rvsdg::ValueType>(outputType);
+    auto linakgeString = mlirDeltaNode.getLinkage().str();
+    auto rvsdgDeltaNode = llvm::delta::node::Create(
+        &rvsdgRegion,
+        outputValueType,
+        mlirDeltaNode.getName().str(),
+        ConvertLinkage(linakgeString),
+        mlirDeltaNode.getSection().str(),
+        mlirDeltaNode.getConstant());
+
+    auto outputVector = ConvertRegion(mlirDeltaNode.getRegion(), *rvsdgDeltaNode->subregion());
+
+    if (outputVector.size() != 1)
+      JLM_UNREACHABLE("Expected 1 output for Delta operation.");
```

Comment:
I think using unreachable here makes sense. Validation can handle printing a nice error message, but the validation code is not even in this repository, so it is nice for us to have a cheap backup test that is never compiled away

Reply:
Yes